File: KnearestElements.java
Executed in Leetcode-Yes
Time Complexity- O(n-k)
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
      if(arr==null||arr.length==0) return new ArrayList<Integer>();
        int high=arr.length-1-k, low=0; int mid=0;
        while(high>low){
            mid= low+(high-low)/2;
            int distLeft= Math.abs(arr[mid]-x);
            int distRight= Math.abs(arr[mid+k]-x);
            if(distLeft>distRight){
                low=mid+1;
            }else
                high=mid-1;
        }
        List<Integer> resukt= new ArrayList<Integer>();
        for(int i=mid;i<mid+k;i++){
            resukt.add(arr[i]);
        }
        return resukt;
    }
}

File: k nearest elements.py
# Time Complexity : O(n-k)    
# Space Complexity : O(1)    
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No

#Optimized Binary Search
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        l = 0
        r = len(arr) - k
        
        
        while l < r:
            mid = (l + r) // 2
            distL = x - arr[mid]
            distR = arr[mid + k] - x
            
            if distL > distR:
                l = mid + 1
                
            else:
                r = mid
                
        return arr[l: l + k]



# Time Complexity : O(nlogn + logk)    
# Space Complexity : O(n)    
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No

#Bruteforce
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        result = []

        for i in range(len(arr)):
            diff = arr[i] - x
            result.append(arr[i])
            
        result.sort()
        #print(result)
        return result[0 : k]
